tools/KoreBuild.Console/Commands/DockerBuildCommand.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace KoreBuild.Console.Commands
+{
+    internal class DockerBuildCommand : SubCommandBase
+    {
+        private const string DockerfileExtension = ".dockerfile";
+        private const string ImagePrefix = "aspnetbuild";
+
+        public CommandArgument Platform { get; set; }
+
+        public List<string> Arguments {get; set; }
+        
+        public string ContainerName { get; set; } = "testcontainer";
+
+        public override void Configure(CommandLineApplication application)
+        {
+            Platform = application.Argument("platform", "The docker platform to run on.");
```

Comment:
I'd suggest using a term like `Image` for this (I realize it makes up only part of the image name, see comment below, but I think it's a clearer name regardless). It's a little more aligned with Docker terminology. Platform makes me thing of Docker for Windows vs Docker for Linux.